Jarvis.Framework.Shared/IdentitySupport/AbstractIdentityTranslator.cs

```diff
@@ -92,18 +97,33 @@ protected String GetAlias(TKey key)
 
         /// <summary>
         /// Return reverse mapping for multiple aliases.
+        /// Returns a Dictionary where the Keys are the requested keys and the value
+        /// is the first mapped value (same behavior as <see cref="GetAlias(TKey)"/>
         /// </summary>
         /// <param name="keys"></param>
         /// <returns>Dictionary with mappings.</returns>
         protected IDictionary<TKey, String> GetAliases(IEnumerable<TKey> keys)
         {
-            if (keys == null || !keys.Any())
+            if (keys?.Any() != true)
                 return new Dictionary<TKey, String>();
 
             return _collection
                 .Find(Builders<MappedIdentity>.Filter.In("AggregateId", keys))
                 .ToEnumerable()
-                .ToDictionary(_ => _.AggregateId, _ => _.ExternalKey);
+                .GroupBy(k => k.AggregateId)
+                .ToDictionary(_ => _.Key, _ => _.First().ExternalKey);
+        }
+
+        /// <summary>
+        /// Returns all the aliases of a given key.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        protected string[] GetAliases(TKey key)

```

Comment:
Non so se è meglio chiamarlo in altro modo ... il GetAliases che accetta una ienumerable torna il dizionario perchè è una traslazione multipla, cosi con un unico elemento il nome della funzione è eguale ma fa una cosa completamente differente ... potrebbe essere il caso di rinominare il vecchio scrivendolo nelle breaking changes che la funzione GetAliases si chiama ora tipo GetAliasForMultipleEntry o qualche cosa del gnere, mentre la GetAliases normale è quela che torna TUTTE le alias di un elemento ....

Reply:
mah... "ForMultipleEntry" mi fa pensare all'argomento (che è un ienumerable" e quindi è già chiaro) più che al valore di ritorno... secondo me basta far funzionare l'ìintellisense... che poi ti dice cosa fa la funzione.